File: src/server/gkill/api/req_res/upload_skill_request.go
package req_res

type UploadSkillRequest struct {
	SessionID  string `json:"session_id"`
	LocaleName string `json:"locale_name"`
	// ZipBase64 は zip の中身。data URI の接頭辞（data:application/zip;base64,）が付いていてもよい。
	ZipBase64 string `json:"zip_base64"`
	// DryRun が true なら置き換えずに、何が起きるかだけを返す（画面の確認の1段目）。
	DryRun bool `json:"dry_run"`
}

File: src/client/classes/api/req_res/upload-skill-request.ts
'use strict'

import { GkillAPIRequest } from '../gkill-api-request'

export class UploadSkillRequest extends GkillAPIRequest {
    // zip_base64 は zip の中身。FileReader.readAsDataURL の結果（data URI）をそのまま入れてよい
    zip_base64: string
    // dry_run が true なら置き換えずに、何が起きるかだけが返る（確認の1段目）
    dry_run: boolean

    constructor() {
        super()
        this.zip_base64 = ""
        this.dry_run = false
    }
}
